pkg/azure/client.go

```diff
@@ -67,43 +68,40 @@ func NewAzClient(subscriptionID SubscriptionID, resourceGroupName ResourceGroup,
 		appGwName:         appGwName,
 		memoizedIPs:       make(map[string]n.PublicIPAddress),
 
-		ctx:        context.Background(),
-		authorizer: authorizer,
+		ctx: context.Background(),
 	}
 
 	if err := az.appGatewaysClient.AddToUserAgent(userAgent); err != nil {
 		glog.Error("Error adding User Agent to App Gateway client: ", userAgent)
 	}
-	az.appGatewaysClient.Authorizer = az.authorizer
-
 	if err := az.publicIPsClient.AddToUserAgent(userAgent); err != nil {
 		glog.Error("Error adding User Agent to Public IP client: ", userAgent)
 	}
-	az.publicIPsClient.Authorizer = az.authorizer
-
 	if err := az.virtualNetworksClient.AddToUserAgent(userAgent); err != nil {
 		glog.Error("Error adding User Agent to Virtual Networks client: ", userAgent)
 	}
-	az.virtualNetworksClient.Authorizer = az.authorizer
-
 	if err := az.subnetsClient.AddToUserAgent(userAgent); err != nil {
 		glog.Error("Error adding User Agent to Subnets client: ", userAgent)
 	}
-	az.subnetsClient.Authorizer = az.authorizer
-
 	if err := az.groupsClient.AddToUserAgent(userAgent); err != nil {
 		glog.Error("Error adding User Agent to Groups client: ", userAgent)
 	}
-	az.groupsClient.Authorizer = az.authorizer
-
 	if err := az.deploymentsClient.AddToUserAgent(userAgent); err != nil {
 		glog.Error("Error adding User Agent to Deployments client: ", userAgent)
 	}
-	az.deploymentsClient.Authorizer = az.authorizer
 
 	return az
 }
 
+func (az *azClient) SetAuthorizer(authorizer autorest.Authorizer) {
+	az.appGatewaysClient.Authorizer = authorizer
```

Comment:
Do we need to check for NULL

Reply:
`ctr` takes care of creating these clients.